contracts/RegisterDomain.sol

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: MIT
```

Comment:
Can modifiers be used?
Can you save gas by avoiding some costly EVM operations like SSTORE?
Does the contract can answer all the questions that are highly probable in terms of data monitoring? -5

Reply:
I still don't see all the functions which support data monitoring, metrics, info. Is there anything else, which is highly required? -3 points